.github/workflows/vulnerability-scanner-tests.yml

```diff
@@ -29,87 +29,94 @@ jobs:
       - name: Calculate deltas
         run: |
 
-          DIFFS=$(git diff --name-only origin/${{ github.head_ref }} origin/${{ github.base_ref }})
+          DIFFS=""
+          
+          # Get the list of modified files
+          if [ -z ${{ env.ACT }} ]; then
+            # CI execution
+            DIFFS=$(git diff --name-only HEAD origin/${{ github.base_ref }})
+            # Local execution
+          else
+            if [ -n "${{ env.REF_BRANCH }}" ]; then
+              # If REF_BRANCH is set, use it to calculate the diff
+              DIFFS=$(git diff --name-only HEAD origin/${{ env.REF_BRANCH }})  
+            else
+              # If REF_BRANCH is not set, use main as base 
+              DIFFS=$(git diff --name-only HEAD origin/main)  
+            fi
+          fi
 
           if echo $DIFFS | grep 'src/shared_modules/router/'; then
             echo "ROUTER=true" >> "$GITHUB_ENV"
-          else
-            echo "ROUTER=false" >> "$GITHUB_ENV"
           fi
 
           if echo $DIFFS | grep 'src/shared_modules/content_manager/'; then
             echo "CONTENT_MANAGER=true" >> "$GITHUB_ENV"
-          else
-            echo "CONTENT_MANAGER=false" >> "$GITHUB_ENV"
           fi
 
           if echo $DIFFS | grep 'src/shared_modules/indexer_connector/'; then
             echo "INDEXER_CONNECTOR=true" >> "$GITHUB_ENV"
-          else
-            echo "INDEXER_CONNECTOR=false" >> "$GITHUB_ENV"
           fi
 
           if echo $DIFFS | grep 'src/wazuh_modules/vulnerability_scanner/'; then
             echo "VULNERABILITY_SCANNER=true" >> "$GITHUB_ENV"
-          else
-            echo "VULNERABILITY_SCANNER=false" >> "$GITHUB_ENV"
           fi
 
       # Router
       - name: Router - Codding style
```

Comment:
I know that it's not part of the scope, but here's a typo related to `Codding style` instead of  `Coding style`

```suggestion
      - name: Router - Coding style
```

> **Note**
> it's introduced in the PR #18318

Reply:
I've fixed all of them